### ISSUE TITLE
Can f(x) be changed during the data generation?

### ISSUE THREAD

**Issue by geohang** [zongyi-li/fourier_neural_operator#56]
Hi, I am a geoscientist. It is a great paper. In the DARCY FLOW part, I wonder if you change f(x) during the data generation, and use this dataset to train, what's the influence on performance. In earth science, the f(x) is usually changed. I don't know if you have tested that. Thank you!

**Comment by zongyi-li** [comment]
Thank you for your interest! Yes you can change f(x) in the data generation. You can also learn the map from f(x) to u(x) using the FNO2d model. We didn't test that, but usually that will be easier compared to the map from a(x) to u(x).

**Comment by geohang** [comment]
Thanks for your reply! That's very helpful. I have another question about "Furthermore since FNO is differentiable, it can easily be applied to PDE-constrained optimization problems without the need for the adjoint method." If I want to have the Jacobian matrix from the FNO to do the optimization, how can I get it? Thank you!

**Comment by zongyi-li** [comment]
The autograd in PyTorch supports the gradient respect to a scaler (loss). In the optimization problem, if you can write out the output of objective function as a scaler, then you can use autograd to take the gradient.

However, if you output is a vector. Say, if FNO maps a 100^2 input vector(images) to a 100^2 output vector(images), the Jacobian matrix will be 100^4. In general you cannot take the jacobian for a neural network with respect to a vector output.

**Comment by geohang** [comment]
Got it! Thank you!
